### PR TITLE
Fix autodoc parameter hrefs

### PR DESCRIPTION
**Pull request by gante**
## TL;DR 

fixes hrefs for class parameters 🤗 

_______________________________________________ 

### Problem

Currently, hrefs for individual parameters are broken on ALL classes across ALL libraries.

Examples (try to click on individual parameters of the classes):
- (transformers) https://huggingface.co/docs/transformers/v4.43.2/en/main_classes/text_generation#transformers.GenerationConfig
- (accelerate) https://huggingface.co/docs/accelerate/v0.33.0/en/package_reference/accelerator

We can see in the pages' source code that the hrefs are sketchy
Screenshot 2024-07-26 at 10 27 11

Diving deeper in the doc builder, we can see that:
1. The creation of the mdx files is correct
2. The issue can be solved if we comment out [the lines](https://github.com/huggingface/doc-builder/blob/fcea09086f6dec68ef9ce419f164c6c33e8fb5f4/kit/preprocessors/hashInCode.js#L12) that replace `#` by their htlm equivalent on inline code blocks

### Solution

The `#` conversion exists to prevent bad formatting in some cases, so it can't simply be removed -- see #504 

The solution adopted consists in detecting inline code blocks inside the autodoc section, which contain the href anchor, and preserving the `#` there. The solution in the PR consists in reverting the conversion -- in my quick experiments, a complete regex that skipped the conversion in autodoc blocks was much much slower.

Example of an href after these changes:
Screenshot 2024-07-26 at 10 37 01

Same href in a [previously working version](https://huggingface.co/docs/transformers/v4.40.2/en/main_classes/text_generation):
Screenshot 2024-07-26 at 10 36 29

### CODE REVIEW

File: kit/preprocessors/hashInCode.js [hunk 1]
```diff
@@ -6,12 +6,17 @@ export const hashInCodePreprocess = {
 	markup: async ({ content }) => {
 		const REGEX_CODE_BLOCK = /```.*?```/gs;
 		const REGEX_INLINE_CODE = /`.*?`/g;
+		const REGEX_INLINE_EXCEPTION = /(?!.*HF_DOC_BODY_START)`.*?`(?<!HF_DOC_BODY_END.*)/gs;
```
Comment (gante) [inline kit/preprocessors/hashInCode.js:9]: logic tl;dr: find all code blocks (``.*?``) between `HF_DOC_BODY_START` and `HF_DOC_BODY_END`, the autodocs delimiters

**Comment by gante** [comment]
@mishig25 ping :)

**Review by mishig25 — APPROVED** [review]
lgtm !
